beacon-chain/core/epoch/participation.go

```diff
@@ -0,0 +1,35 @@
+package epoch
+
+import (
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	ethpb "github.com/prysmaticlabs/prysm/proto/eth/v1alpha1"
+)
+
+// ComputeValidatorParticipation by matching validator attestations during the epoch,
+// computing the attesting balance, and how much attested compared to the total balances.
+func ComputeValidatorParticipation(state *pb.BeaconState) (*ethpb.ValidatorParticipation, error) {
+	currentEpoch := helpers.SlotToEpoch(state.Slot)
+	finalized := currentEpoch == state.FinalizedCheckpoint.Epoch
```

Comment:
this will always be false. The current epoch can never be finalized, due to how finalization works